src/os/FileJournal.cc

```diff
@@ -1719,6 +1721,9 @@ bool FileJournal::read_entry(
     &ss);
   if (result == SUCCESS) {
     if (next_seq > seq) {
+      if (next_seq - 1 > journaled_seq) {
+        journaled_seq = next_seq -1;
+      }
       return false;
     } else {
       read_pos = next_pos;
```

Comment:
say we replay events 90 through 99... we'll get true from read_entry 10 times (90...99) and then fails on 100.  i think we should be setting journaled_seq = seq ( = next_seq) on each of those successful reads, and then do nothing on the final failure?